src/equiv.js

```diff
@@ -12,8 +12,11 @@ QUnit.equiv = (function() {
 	function useStrictEquality( b, a ) {
 
 		/*jshint eqeqeq:false */
-		if ( b instanceof a.constructor || a instanceof b.constructor ) {
+		if ( b instanceof a.constructor && a instanceof b.constructor ) {
 
+			// Compare same constructor.
+			return a.toString() == b.toString();
```

Comment:
I think this could be generalized and simplified a bit further, to something like
```js
if ( typeof a === "object" ) {
	a = a.valueOf();
}
if ( typeof b === "object" ) {
	b = b.valueOf();
}
return a === b;
```